File: 2340.minimum-adjacent-swaps-to-make-a-valid-array.py
from typing import *
class Solution:
    def minimumSwaps(self, nums: List[int]) -> int:
        smallest = nums[0]
        largest = nums[-1]
        
        min_small_index = 0
        max_large_index = len(nums) - 1
        
        for i in range(len(nums)):
          if nums[i] < smallest:
            min_small_index = i
            smallest = nums[i]
          if nums[len(nums) - 1 - i] > largest:
            max_large_index = len(nums) - 1 - i
            largest = nums[len(nums) - 1 - i]
        
        ans = min_small_index - 0 + len(nums) - 1 - max_large_index
        
        if min_small_index > max_large_index:
          ans -= 1
        
        return ans
        

File: 2340.minimum-adjacent-swaps-to-make-a-valid-array.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution
{
public:
  int minimumSwaps(vector<int> &nums)
  {
    int smallest = nums.front(),
        largest = nums.back(),
        first_small_index = 0,
        first_large_index = nums.size() - 1;

    for (int i = 0; i < nums.size(); ++i)
    {
      if (nums[i] < smallest)
      {
        smallest = nums[i];
        first_small_index = i;
      }

      if (nums[nums.size() - 1 - i] > largest)
      {
        largest = nums[nums.size() - 1 - i];
        first_large_index = nums.size() - 1 - i;
      }
    }

    return first_small_index + nums.size() - 1 - first_large_index - (int)(first_small_index > first_large_index);
  }
};